src/base/server_type.cpp

```diff
@@ -201,6 +208,73 @@ void setup_tran_server_params_on_single_node_config ()
   prm_set_bool_value (PRM_ID_REMOTE_STORAGE, true);
 }
 
+int setup_tran_server_params_on_ha_mode ()
+{
+  assert (!HA_DISABLED ());
+
+  char *page_hosts_new_value = NULL;
+  constexpr size_t MAX_BUFSIZE = 4096;
+  int new_value_size = 0;
+
+  char ha_node_list[MAX_BUFSIZE];
+  char *str, *savep;
+
+  int port_id = prm_get_master_port_id ();
+
+  page_hosts_new_value = (char *) calloc (MAX_BUFSIZE, sizeof (char)); // free is called by sysprm_final()
+  if (page_hosts_new_value == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, MAX_BUFSIZE);
+      return ER_OUT_OF_VIRTUAL_MEMORY;
+    }
+  new_value_size = MAX_BUFSIZE;
+
+  strncpy_bufsize (ha_node_list, prm_get_string_value (PRM_ID_HA_NODE_LIST));
+
+  str = strtok_r (ha_node_list, "@", &savep); // dbname@host1:host2:...
+  str = strtok_r (NULL, " ,:", &savep);
+  while (str)
+    {
+      char tmp[MAX_BUFSIZE] = {0};
+      sprintf (tmp, "%s:%d,", str, port_id);
```

Comment:
Does it mean that all PSes have to use the same port as the ATS's?

Reply:
Yes, it can be a restriction.
In oldHA, it was also required for all nodes to have the same value for the port ID (same master port id)
While this issue could be improved in newHA, but I think it is better to not make any modifications for now.